### PR TITLE
neovide: fix build for aarch64-darwin

### PR DESCRIPTION
**Pull request by n8henrie**
Thanks to @wegank for https://github.com/NixOS/nixpkgs/pull/274839/files

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/editors/neovim/neovide/default.nix [hunk 1]
```diff
@@ -81,6 +81,11 @@ rustPlatform.buildRustPackage.override { stdenv = clangStdenv; } rec {
     darwin.apple_sdk.frameworks.AppKit
   ];
 
+ env = lib.optionalAttrs stdenv.isDarwin {
```
Comment (wegank) [inline pkgs/applications/editors/neovim/neovide/default.nix:84]:
```suggestion
  env = lib.optionalAttrs stdenv.isDarwin {
```

**Review by MultisampledNight — APPROVED** [review]
Thank you! :sparkles:

**Comment by nekowinston** [comment]
Result of `nixpkgs-review pr 279185` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>neovide</li>
  </ul>
</details>

**Review by ck3d — APPROVED** [review]
LGTM, thanks for fixing

**Comment by n8henrie** [comment]
It seems that checks can be reenabled by adding neovim to the check dependencies.

**Comment by ck3d** [comment]
Could you please cleanup your commits

**Comment by n8henrie** [comment]
Of course. It also looks like x86-64 darwin is failing, I'm investigating whether I can debug this cross compiling from aarch64.

**Review by thiagokokada — APPROVED** [review]

**Comment by pitkling** [comment]
> Of course. It also looks like x86-64 darwin is failing, I'm investigating whether I can debug this cross compiling from aarch64.

Just wondering @n8henrie, was the problem on x86 darwin resolved before this got merged? On my aarch64 mac this works perfectly, but the same system configuration fails to build on my x86-64 mac (because of neovide).

**Comment by n8henrie** [comment]
@pitkling No, it wasn't! I'm on aarch64-darwin so I wasn't sure if it was a cross problem or not, thanks for confirming that it's an issue on x86_64.

I was getting an error about `no matching function for call to 'min'`.

**Comment by pitkling** [comment]
> I was getting an error about `no matching function for call to 'min'`.

Thanks for confirming. I'm pretty sure that's the same error I got. I will check and post the log in a new issue, linking this one. Probably not before next week, though.

**Pull request merged**